src/utils/power_flow/make_pf.jl

```diff
@@ -0,0 +1,142 @@
+"""
+    make_pf(sys)
+
+Create the objects needed to solve an powerflow case using NLsolve.jl solvers. Returns 
+an anonymous function with the powerflow equations, initial conditions and a dict to link the
+solutions to the original system. Only supports systems with a single generator per bus and 
+currently doesn't support distributes slack buses and doesn't enforce reactive power limits. 
+
+## Example
+```julia
+pf!, x0, res_ref = make_pf(sys)  
+res = NLsolve.nlsolve(pf!, x0)
+```
+
+# Arguments
+    * `sys`::System : a PowerSystems.jl system
+
+"""
+
+macro fn(expr::Expr)
+    @assert expr.head in (:function, :->)
+    name = gensym()
+    args = expr.args[1]
+    args = typeof(args) == Symbol ? [args] : args.args
+    body = expr.args[2]
+    @eval $name($(args...)) = $body
+    name
+  end
+
+function make_pf(system)
+    buses = sort(collect(get_components(Bus, system)), by = x -> x.number)
```

Comment:
This is likely to be a common activity, right?  If so, we should make a function and possibly export it.

Reply:
Ok, I will add this as a function. Good call.